### PR TITLE
User profile

### PR DESCRIPTION
**Pull request by anasm11**
added user profile and follow/unfollow functionality

Main files to check: 
Profile.jsx
User.jsx

### CODE REVIEW

File: src/pages/User.jsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                    <div className='footer flex items-center justify-around'>
+                        <div className="action-icons items-center flex gap-4">
+                            <span className="flex items-center">
+                                {post.likes.likedBy.find((person) => person.firstName === 'a') ? <IcRoundFavorite /> : <IcRoundFavoriteBorder onClick={() => likeHandler(post)} />}
+                                {post.likes.likeCount}
+                            </span>
+
+                        </div>
+                        {selfname===username &&<div className='flex gap-4'>
+                        {/* <button className='button' onClick={() => { setShowEditModal(post) }}>Edit</button>
```
Comment (kaustubh-jsr) [inline src/pages/User.jsx]: You can remove this comment, if you are not going to use this button

File: src/pages/Profile.jsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                {isEditState ?
+                    <label for='input-url'>
+                        <input type="url" name='input-url' value={url} onChange={(e) => { setURL(e.target.value) }} class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-600 dark:border-gray-500 dark:placeholder-gray-400 dark:text-white" placeholder="url" required />
+                    </label>
+                    : <a className='link'>{url}</a>}
+            </div>
+            <button className='button' onClick={async()=>{
+                setIsEditState(false)
+
+                const res=await axios({
```
Comment (kaustubh-jsr) [inline src/pages/Profile.jsx]: You can abstract out this inline function call too, will make the code more readable

File: src/pages/Profile.jsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            setUser(user)
+        })()
+    },[])
+
+    return (<div className='flex relative gap-5'>
+        <input
+        type="file"
+        name="myImage"
+        ref={inputFile}
+        onChange={(event) => {
```
Comment (kaustubh-jsr) [inline src/pages/Profile.jsx]: This function can be abastracted out, rather than keeping it in line.

**Review by kaustubh-jsr — COMMENTED** [review]
The code looks good, suggested some minor tweaks. Rest of the code looks clean.

File: src/pages/Profile.jsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            }
+            reader.readAsDataURL(event.target.files[0]);
+        }}
+        style={{display: 'none'}}
+      />
+        <span className='edit-photo-container bg-gray-300 flex' onClick={(event) => {
+            inputFile.current.click()            
+          }}>
+        
+            <img className='edit-profile-photo' width={'200px'} src={image} />
```
Comment (yaswanthmyneni) [inline src/pages/Profile.jsx]: changes: use alt attribute for images.

**Review by yaswanthmyneni — APPROVED** [review]
the code is good, you can merge after a change

**Pull request merged**